Sources/Basic/TemporaryFile.swift

```diff
@@ -0,0 +1,172 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+
+ This file provides API to create and use temporary files and directories.
+*/
+
+import libc
+import func POSIX.getenv
+import class Foundation.NSFileHandle
+import var Foundation.NSUTF8StringEncoding
+import func Utility.fputs // FIXME: we should move file related helper methods to Basic.
```

Comment:
Agreed.  I have put up a PR for the path class (which is in Basic), and will be removing the Basic -> Utility dependency once it lands.